openapi/swagger: Conflicting setter definitions for property

After upgrading to latest swagger-parser: 2.0.23 the following error occurs:

Stacktrace:
```
Caused by: java.lang.IllegalArgumentException: Conflicting setter definitions for property "protocol": io.jooby.SslOptions#setProtocol(1 params) vs io.jooby.SslOptions#setProtocol(1 params)
	at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.getSetter(POJOPropertyBuilder.java:504)
	at io.swagger.v3.core.jackson.ModelResolver.ignore(ModelResolver.java:995)
	at io.swagger.v3.core.jackson.ModelResolver.resolve(ModelResolver.java:572)
	at io.swagger.v3.core.jackson.AbstractModelConverter.resolve(AbstractModelConverter.java:52)
	at io.jooby.internal.openapi.ModelConverterExt.resolve(ModelConverterExt.java:42)
	at io.swagger.v3.core.jackson.AbstractModelConverter.resolve(AbstractModelConverter.java:52)
	at io.jooby.internal.openapi.ModelConverterExt.resolve(ModelConverterExt.java:42)
```

Problem is on multiple definitions of `SslOptions.setProtocol` (still not sure why parser reach this class).

This bug isn't part of any release, but build is failing because of this error.

--- a/modules/jooby-openapi/pom.xml
+++ b/modules/jooby-openapi/pom.xml
@@ -57,6 +57,12 @@
     </dependency>
 
     <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+      <version>${jackson.version}</version>
+    </dependency>
+
+    <dependency>
       <groupId>io.swagger.core.v3</groupId>
       <artifactId>swagger-models</artifactId>
     </dependency>

--- a/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/ConflictiveSetter.java
+++ b/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/ConflictiveSetter.java
@@ -1,0 +1,47 @@
+package io.jooby.internal.openapi;
+
+import java.util.List;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod;
+
+/**
+ * Fix for:
+ *
+ * Caused by: java.lang.IllegalArgumentException: Conflicting setter definitions for property
+ * 	at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.getSetter(POJOPropertyBuilder.java:504)
+ * 	at io.swagger.v3.core.jackson.ModelResolver.ignore(ModelResolver.java:995)
+ * 	at io.swagger.v3.core.jackson.ModelResolver.resolve(ModelResolver.java:572)
+ */
+class ConflictiveSetter extends AnnotationIntrospector {
+  @Override public Version version() {
+    return Version.unknownVersion();
+  }
+
+  @Override
+  public AnnotatedMethod resolveSetterConflict(MapperConfig<?> config, AnnotatedMethod setter1,
+      AnnotatedMethod setter2) {
+    Class<?> cls1 = setter1.getRawParameterType(0);
+    Class<?> cls2 = setter2.getRawParameterType(0);
+    if (isArrayLike(cls1, cls2)) {
+      return setter1;
+    }
+    if (isArrayLike(cls2, cls1)) {
+      return setter2;
+    }
+    return null;
+  }
+
+  /**
+   * Returns true if type1 is a list and type2 an array.
+   *
+   * @param type1 Type 1.
+   * @param type2 Type 2.
+   * @return True if type1 is a list and type2 an array.
+   */
+  private boolean isArrayLike(Class type1, Class type2) {
+    return List.class.isAssignableFrom(type1) && type2.isArray();
+  }
+}

--- a/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/ParserContext.java
+++ b/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/ParserContext.java
@@ -5,43 +5,10 @@
  */
 package io.jooby.internal.openapi;
 
-import com.fasterxml.jackson.databind.JavaType;
-import com.fasterxml.jackson.databind.Module;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
-import io.jooby.Context;
-import io.jooby.FileUpload;
-import io.jooby.SneakyThrows;
-import io.jooby.StatusCode;
-import io.jooby.openapi.DebugOption;
-import io.swagger.v3.core.converter.ModelConverters;
-import io.swagger.v3.core.converter.ResolvedSchema;
-import io.swagger.v3.core.util.Json;
-import io.swagger.v3.core.util.RefUtils;
-import io.swagger.v3.core.util.Yaml;
-import io.swagger.v3.oas.models.media.ArraySchema;
-import io.swagger.v3.oas.models.media.BinarySchema;
-import io.swagger.v3.oas.models.media.BooleanSchema;
-import io.swagger.v3.oas.models.media.ByteArraySchema;
-import io.swagger.v3.oas.models.media.FileSchema;
-import io.swagger.v3.oas.models.media.IntegerSchema;
-import io.swagger.v3.oas.models.media.MapSchema;
-import io.swagger.v3.oas.models.media.NumberSchema;
-import io.swagger.v3.oas.models.media.ObjectSchema;
-import io.swagger.v3.oas.models.media.Schema;
-import io.swagger.v3.oas.models.media.StringSchema;
-import io.swagger.v3.oas.models.media.UUIDSchema;
-import org.objectweb.asm.ClassReader;
-import org.objectweb.asm.ClassVisitor;
-import org.objectweb.asm.Opcodes;
-import org.objectweb.asm.Type;
-import org.objectweb.asm.tree.AbstractInsnNode;
-import org.objectweb.asm.tree.ClassNode;
-import org.objectweb.asm.tree.MethodNode;
-import org.objectweb.asm.util.ASMifier;
-import org.objectweb.asm.util.Printer;
-import org.objectweb.asm.util.TraceClassVisitor;
-import org.objectweb.asm.util.TraceMethodVisitor;
+import static io.jooby.internal.openapi.TypeFactory.COROUTINE_ROUTER;
+import static io.jooby.internal.openapi.TypeFactory.JOOBY;
+import static io.jooby.internal.openapi.TypeFactory.KOOBY;
+import static io.jooby.internal.openapi.TypeFactory.ROUTER;
 
 import java.io.File;
 import java.io.InputStream;
@@ -71,10 +38,45 @@ import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static io.jooby.internal.openapi.TypeFactory.COROUTINE_ROUTER;
-import static io.jooby.internal.openapi.TypeFactory.JOOBY;
-import static io.jooby.internal.openapi.TypeFactory.KOOBY;
-import static io.jooby.internal.openapi.TypeFactory.ROUTER;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.util.ASMifier;
+import org.objectweb.asm.util.Printer;
+import org.objectweb.asm.util.TraceClassVisitor;
+import org.objectweb.asm.util.TraceMethodVisitor;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import io.jooby.Context;
+import io.jooby.FileUpload;
+import io.jooby.SneakyThrows;
+import io.jooby.StatusCode;
+import io.jooby.openapi.DebugOption;
+import io.swagger.v3.core.converter.ModelConverters;
+import io.swagger.v3.core.converter.ResolvedSchema;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.core.util.RefUtils;
+import io.swagger.v3.core.util.Yaml;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.BinarySchema;
+import io.swagger.v3.oas.models.media.BooleanSchema;
+import io.swagger.v3.oas.models.media.ByteArraySchema;
+import io.swagger.v3.oas.models.media.FileSchema;
+import io.swagger.v3.oas.models.media.IntegerSchema;
+import io.swagger.v3.oas.models.media.MapSchema;
+import io.swagger.v3.oas.models.media.NumberSchema;
+import io.swagger.v3.oas.models.media.ObjectSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.media.StringSchema;
+import io.swagger.v3.oas.models.media.UUIDSchema;
 
 public class ParserContext {
 
@@ -121,6 +123,13 @@ public class ParserContext {
     } catch (ClassNotFoundException | InstantiationException | IllegalAccessException x) {
       // Sshhhhh
     }
+    /** Ignore some conflictive setter in Jooby API: */
+    modules.add(new SimpleModule("jooby-openapi") {
+      @Override public void setupModule(SetupContext context) {
+        super.setupModule(context);
+        context.insertAnnotationIntrospector(new ConflictiveSetter());
+      }
+    });
     /** Java8/Optional: */
     modules.add(new Jdk8Module());
     modules.forEach(module -> mappers

